File: taxis.jl
using CSV, DataFrames

@time taxis = [
    CSV.read("yellow_tripdata_2020-01.csv", DataFrame);
    CSV.read("yellow_tripdata_2020-02.csv", DataFrame);
    CSV.read("yellow_tripdata_2020-03.csv", DataFrame);
    CSV.read("yellow_tripdata_2020-04.csv", DataFrame);
]
df = DataFrame(:a => [1;2], :b => [2;3])
transform!(df, [:a,:b] => ((x,y)->x.*y))

combine(groupby(df, :a), nrow, :a => sum)

@time taxis |> 
  x -> transform!(x,[:passenger_count,:trip_distance] => ((x,y)->x.*y) => :total_distance) |> 
  x -> groupby(x, :passenger_count) |> 
  x -> combine(x, nrow, :total_distance => sum)

File: taxis.R
setwd("~/Documents/JuliaR")
library(tidyverse)
library(tictoc)

tic("read data")
taxi <- bind_rows(read_csv("yellow_tripdata_2020-01.csv"),
                  read_csv("yellow_tripdata_2020-02.csv"),
                  read_csv("yellow_tripdata_2020-03.csv"),
                  read_csv("yellow_tripdata_2020-04.csv")
)
toc()
tic("summarise data")
taxi %>% 
  mutate(total_distance = passenger_count*trip_distance) %>% 
  select(total_distance, passenger_count, trip_distance) %>%
  group_by(passenger_count) %>%
  summarise(n = n(), total = sum(total_distance))
toc()